content/file/file_test.go

```diff
@@ -391,7 +391,7 @@ func TestStore_Dir_Push(t *testing.T) {
 	if err != nil {
 		t.Fatal("failed to open internal gz")
 	}
-	gz, err := io.ReadAll(zrc)
+	gz, err := content.ReadAll(zrc, desc)
```

Comment:
Please take a close look at this test case, TestStore_Dir_Push, as I replaced some but not all io.ReadAll so that the test can pass.

Reply:
`content.ReadAll()` should only applies to `io.ReadCloser`s obtained from `Fetch()` or its equivalents.

Reply:
This line has nothing to do with descriptors, so `content.ReadAll()` is not applicable here.